build-tools/packages/build-tools/src/common/biomeConfig.ts

```diff
@@ -0,0 +1,203 @@
+ */
+export async function getSettingValuesFromBiomeConfig(
+	config: BiomeConfig,
+	section: BiomeConfigSection,
+	kind: BiomeIncludeIgnore,
+): Promise<Set<string>> {
+	if (section === "formatter" && kind === "ignore") {
+		const filesIgnore = config.files?.ignore ?? [];
+		const formatterIgnores = config.formatter?.ignore ?? [];
+		return new Set([...filesIgnore, ...formatterIgnores]);
+	}
+
+	if (section === "formatter" && kind === "include") {
+		const filesInclude = config.files?.include ?? [];
+		const formatterIncludes = config.formatter?.include ?? [];
+		return new Set([...filesInclude, ...formatterIncludes]);
+	}
+
+	if (section === "linter" && kind === "ignore") {
+		const filesIgnore = config.files?.ignore ?? [];
+		const linterIgnores = config.linter?.ignore ?? [];
+		return new Set([...filesIgnore, ...linterIgnores]);
+	}
+
+	if (section === "linter" && kind === "include") {
+		const filesInclude = config.files?.include ?? [];
+		const linterIncludes = config.linter?.include ?? [];
+		return new Set([...filesInclude, ...linterIncludes]);
+	}
+
+	return new Set<string>();
+}
+
+/**
+ * Returns the absolute path to the closest Biome config file found from the current working directory up to the root
+ * of the repo.
+ */
+export async function getClosestBiomeConfigPath(
+	cwd: string,
+	stopAt?: string,
+): Promise<string | undefined> {
+	return (await findUp)
+		.findUp(["biome.json", "biome.jsonc"], { cwd, stopAt })
+		.then((config) => {
+			if (config === undefined) {
+				throw new Error(`Can't find biome config file`);
+			}
+			return config;
+		});
+}
+
+/**
+ * Return an array of absolute paths to files that Biome would format under the provided path. Note that .gitignored
+ * paths are always excluded, regardless of the "vcs" setting in the Biome configuration.
+ *
+ * @param directoryOrConfigFile - A path to a directory or a Biome config file. If a directory is provided, then the
+ * closest Biome configuration will be loaded and used. If a path to a file is provided, it is assumed to be a Biome
+ * config file and will be loaded as such. The directory containing the config file will be used as the working
+ * directory when applying the Biome include/ignore settings.
+ * @param gitRepo - A GitRepo instance that is used to enumerate files.
+ */
+export async function getBiomeFormattedFiles(
+	directoryOrConfigFile: string,
+	gitRepo: GitRepo,
+): Promise<string[]> {
+	let configFile: string | undefined;
+	let directory: string;
+
+	if ((await stat(directoryOrConfigFile)).isFile()) {
+		configFile = directoryOrConfigFile;
+		directory = path.dirname(directoryOrConfigFile);
+	} else {
+		configFile = await getClosestBiomeConfigPath(directoryOrConfigFile);
+		directory = directoryOrConfigFile;
+	}
+
+	/**
+	 * All files that could possibly be formatted before ignore entries are applied. Paths are relative to the root of
+	 * the repo.
+	 */
+	const allPossibleFiles = new Set(await gitRepo.getFiles(directory));
```

Comment:
I feel like git would let you specify globs rather than getting all and post-fixing. At least should be able to give it the included set.

Reply:
Good point.

Reply:
I looked into this option a bit more, and while it seems like it should work, the code got pretty ugly, and I broke a couple tests.

In the latest iteration, I replaced the globby code with an equivalent library that applies globs to a list of paths. So now the flow is:

git ls-files --> if config has `includes`, apply those globs to filter the list, otherwise pass all paths through --> apply `ignores` --> list of paths to put in cache